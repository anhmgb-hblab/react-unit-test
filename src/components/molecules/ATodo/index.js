import React from 'react';
import { DeleteFilled } from '@ant-design/icons';
import PropTypes from 'prop-types';
import { ACheckbox, AButton } from '../../atoms';
import StyledTodo from './styled';

const ATodo = ({
  content,
  checked,
  onChangeStatus,
  onDeleteTodo,
}) => (
  <StyledTodo className="a-todo">
    <div className="left-content">
      <ACheckbox checked={checked} onChange={onChangeStatus} />
      <div className="content">
        {content}
      </div>
    </div>
    <AButton
      type="text"
      icon={<DeleteFilled />}
      onClick={onDeleteTodo}
    />
  </StyledTodo>
);

ATodo.defaultProps = {
  content: '',
  checked: false,
  onChangeStatus: null,
  onDeleteTodo: null,
};

ATodo.propTypes = {
  content: PropTypes.string,
  checked: PropTypes.bool,
  onChangeStatus: PropTypes.func,
  onDeleteTodo: PropTypes.func,
};

export default ATodo;
