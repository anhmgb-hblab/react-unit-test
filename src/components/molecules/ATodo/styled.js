import styled from 'styled-components';
import { COLORS, BORDER_RADIUS } from '../../../constants/styles';

export default styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 8px;
  background-color: ${COLORS.TODO};
  max-width: 500px;
  border-radius: ${BORDER_RADIUS};
`;
