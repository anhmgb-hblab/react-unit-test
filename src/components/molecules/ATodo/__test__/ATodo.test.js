import React from 'react';
import toJson from 'enzyme-to-json';
import { mount } from 'enzyme';
import ATodo from '../index';

describe('Test ATodo running correctly', () => {
  it('test todo props', () => {
    const wrapper = mount(<ATodo content="Todo" />);
    expect(wrapper.props().checked).toEqual(false);
    expect(wrapper.props().content).toEqual('Todo');
  });

  it('test rendering', () => {
    const tree = mount(<ATodo content="Todo" />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
