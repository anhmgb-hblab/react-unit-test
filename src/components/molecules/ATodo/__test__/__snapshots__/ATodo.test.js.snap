// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ATodo running correctly test rendering 1`] = `
<ATodo
  checked={false}
  content="Todo"
  onChangeStatus={null}
  onDeleteTodo={null}
>
  <styled.div
    className="a-todo"
  >
    <div
      className="sc-gsTEea dHHdsV a-todo"
    >
      <div
        className="left-content"
      >
        <ACheckbox
          checked={false}
          disabled={false}
          label={null}
          onChange={null}
        >
          <Checkbox
            checked={false}
            disabled={false}
            onChange={null}
          >
            <label
              className="ant-checkbox-wrapper"
            >
              <Checkbox
                checked={false}
                className=""
                defaultChecked={false}
                disabled={false}
                onBlur={[Function]}
                onChange={null}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyPress={[Function]}
                onKeyUp={[Function]}
                prefixCls="ant-checkbox"
                style={Object {}}
                type="checkbox"
              >
                <span
                  className="ant-checkbox"
                  style={Object {}}
                >
                  <input
                    checked={false}
                    className="ant-checkbox-input"
                    disabled={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyPress={[Function]}
                    onKeyUp={[Function]}
                    type="checkbox"
                  />
                  <span
                    className="ant-checkbox-inner"
                  />
                </span>
              </Checkbox>
              <span />
            </label>
          </Checkbox>
        </ACheckbox>
        <div
          className="content"
        >
          Todo
        </div>
      </div>
      <AButton
        disabled={false}
        icon={<ForwardRef(DeleteFilled) />}
        loading={false}
        onClick={null}
        size="middle"
        type="text"
      >
        <Button
          disabled={false}
          icon={<ForwardRef(DeleteFilled) />}
          loading={false}
          onClick={null}
          size="middle"
          type="text"
        >
          <button
            className="ant-btn ant-btn-text ant-btn-icon-only"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <ForwardRef(DeleteFilled)>
              <AntdIcon
                icon={
                  Object {
                    "icon": Object {
                      "attrs": Object {
                        "focusable": "false",
                        "viewBox": "64 64 896 896",
                      },
                      "children": Array [
                        Object {
                          "attrs": Object {
                            "d": "M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z",
                          },
                          "tag": "path",
                        },
                      ],
                      "tag": "svg",
                    },
                    "name": "delete",
                    "theme": "filled",
                  }
                }
              >
                <span
                  aria-label="delete"
                  className="anticon anticon-delete"
                  role="img"
                >
                  <IconReact
                    icon={
                      Object {
                        "icon": Object {
                          "attrs": Object {
                            "focusable": "false",
                            "viewBox": "64 64 896 896",
                          },
                          "children": Array [
                            Object {
                              "attrs": Object {
                                "d": "M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z",
                              },
                              "tag": "path",
                            },
                          ],
                          "tag": "svg",
                        },
                        "name": "delete",
                        "theme": "filled",
                      }
                    }
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="delete"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      key="svg-delete"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
                        key="svg-delete-svg-0"
                      />
                    </svg>
                  </IconReact>
                </span>
              </AntdIcon>
            </ForwardRef(DeleteFilled)>
          </button>
        </Button>
      </AButton>
    </div>
  </styled.div>
</ATodo>
`;
